pkg/api/types.go

```diff
@@ -808,7 +809,11 @@ func (p *Properties) GetAgentVMPrefix(a *AgentPoolProfile) string {
 	vmPrefix := ""
 	if index != -1 {
 		if a.IsWindows() {
-			vmPrefix = nameSuffix[:4] + p.K8sOrchestratorName() + fmt.Sprintf("%02d", index)
+			if a.NewWindowsVMNaming != nil && *a.NewWindowsVMNaming {
```

Comment:
Using my suggestion this would change to:

`if a.NameVersion == "v2" {`

@xizhamsft @CecileRobertMichon  thoughts?

Reply:
I like that, it's less binary and gives us more room for future changes

Reply:
Do we need to differentiate between windows name version vs linux name version? Chances are linux name version never need to be changed. So this should be more like "windowsNameVersion"?

Reply:
Yes, it should be `WindowsNameVersion` (`windowsNameVersion` in JSON)